homeassistant/components/hive/config_flow.py

```diff
@@ -106,7 +105,12 @@ async def async_step_finish(self):
                 await self.hass.config_entries.async_reload(self.entry.entry_id)
                 return self.async_abort(reason="reauth_successful")
             return self.async_create_entry(title=self.data["username"], data=self.data)
-        return self.async_abort(reason="unknown")
+
+        errors["base"] = "unknown"
+        schema = vol.Schema(
+            {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+        )
+        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
```

Comment:
Don't move directly to another step's form. And don't pass errors between steps. Errors are specific to the form where they came from.

Since the finish step doesn't have a form, it's better to make it a helper method instead of a step method. Raise the errors encountered and catch them in the calling step method.